presto-main/src/main/java/com/facebook/presto/operator/MergeHashSort.java

```diff
@@ -45,11 +45,23 @@ public MergeHashSort(AggregatedMemoryContext memoryContext)
     /**
      * Rows with same hash value are guaranteed to be in the same result page.
      */
-    public WorkProcessor<Page> merge(List<Type> keyTypes, List<Type> allTypes, List<WorkProcessor<Page>> channels, DriverYieldSignal driverYieldSignal)
+    public WorkProcessor<Page> merge(List<Type> keyTypes, List<Type> allTypes, List<WorkProcessor<Page>> pages, DriverYieldSignal driverYieldSignal)
     {
-        InterpretedHashGenerator hashGenerator = InterpretedHashGenerator.createPositionalWithTypes(keyTypes);
+        return merge(keyTypes, null, allTypes, pages, driverYieldSignal);
+    }
+
+    public WorkProcessor<Page> merge(List<Type> keyTypes, List<Integer> keyChannels, List<Type> allTypes, List<WorkProcessor<Page>> pages, DriverYieldSignal driverYieldSignal)
+    {
+        InterpretedHashGenerator hashGenerator;
+        if (keyChannels == null) {
```

Comment:
what are the keyChannels, and why doesn't everything have it?

Reply:
Documenting the reasoning i provided offline.

InterpretedHashGenerator has 2 ways of instantiation 
1. Specify only `hashTypes` - in which case its assumes that the first N columns/channels are assumed to be the `hashChannels`. HashAggregations fit this criteria and hence SpillableHashAggregationBuilder instantiates it through this approach

2. Specify `hashTypes`, `hashChannels` - This is used by TopN, because the first N columns are not necessarily the grouping keys, so we specify explicitly